pkg/operator/ceph/pool/radosnamespace/controller.go

```diff
@@ -261,6 +261,18 @@ func (r *ReconcileCephBlockPoolRadosNamespace) reconcile(request reconcile.Reque
 		return reconcile.Result{}, errors.Wrap(err, "failed to save cluster config")
 	}
 
+	if cephBlockPoolRadosNamespace.Spec.Mirroring.Enabled {
```

Comment:
we need to check if the mirroring is enabled on the blockPool as it's the base requirement or else this command always fails (we can also depend on the ceph for it instead of we introducing these checks).

Reply:
if we add this kind of check we also need to ensure that mirroring at the pool level is disabled only after all the namespace-level mirroring's for this pool are disabled.

Reply:
Currently We are disabling using the cmd `rbd mirror pool disable` which is same for cephblockpool, 
Taking more input from ceph team,
If the command is same, probably good to just keep this disable function at cephblockpool controller

Reply:
@Madhu-1 we need to test it explicitly ceph enables rn even the pool mirroring is disabled

Reply:
> if we add this kind of check we also need to ensure that mirroring at the pool level is disabled only after all the namespace-level mirroring's for this pool are disabled.

Same here. Disabling mirroring on a pool requires disabling mirroring on images in the default namespace and at the very least also removing mirror peers on both sides, but if one does all that I don't think there is any explicit check in RBD that would prevent mirroring from being disabled on a pool even if it's still enabled on one or more of the custom namespaces.

Reply:
Correct @idryomov I have tested it, and added these checks in the second commit of this PR.
Thanks

Plus I saw a bug for disabling pool mirroring while not disabling the rados-namesapce mirrroing(added it to the design doc)

```
Operation:
1) I enabled the mirroring on test(pool) and r1(namespace)
2) then I disabled the mirroring in test(pool)
3) If I do rbd mirror pool info test/r1 I don't see any output
```

Reply:
> I think it's because at step 3 it would fail to look up the mirror UUID

Yaa correct, as the default namespace would be disabled by disabling the pool level mirroring, other namespace should not work.